chunky/src/test/se/llbit/chunky/renderer/renderdump/RenderDumpTests.java

```diff
@@ -0,0 +1,156 @@
+/* Copyright (c) 2021 Chunky contributors
+ *
+ * This file is part of Chunky.
+ *
+ * Chunky is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chunky is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.llbit.chunky.renderer.renderdump;
+
+import org.junit.Test;
+import se.llbit.chunky.renderer.scene.Scene;
+import se.llbit.util.ProgressListener;
+import se.llbit.util.TaskTracker;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class RenderDumpTests {
+  protected static final int testWidth = Scene.MIN_CANVAS_WIDTH;
+  protected static final int testHeight = Scene.MIN_CANVAS_HEIGHT;
+  protected static final int testSPP = 100;
+  protected static final long testRenderTime = 654321L;
+
+  protected static final double[][][] humanReadable1QuarterRGBSamples = {
+    {{1, 1, 1}, {0, 0, 0}, {1, 1, 1}, {1, 0, 0}, {0, 1, 0}},
+    {{0, 0, 0}, {1, 1, 1}, {0, 0, 0}, {0, 1, 0}, {0, 0, 1}},
+    {{1, 1, 1}, {0, 0, 0}, {1, 1, 1}, {0, 0, 1}, {1, 0, 0}},
+    {{1, 1, 0}, {1, 0, 1}, {0, 1, 1}, {0, 0.25, 0.5}, {0.125, 0.1, 0.5}},
+    {{2, 0.01, 0}, {2, 0, 0.01}, {0, 0.01, 2}, {0.5, 0.1, 0.25}, {0.5, 0.25, 0}}
+  };
+  protected static final double[] testSampleBuffer;
+
+  static {
+    int rgbSamplesHeight = humanReadable1QuarterRGBSamples.length;
+    int rgbSamplesWidth = humanReadable1QuarterRGBSamples[0].length;
+    testSampleBuffer = new double[testWidth * testHeight * 3];
+    int index;
+    for (int y = 0; y < testWidth; y++) {
+      for (int x = 0; x < testHeight; x++) {
+        double[] rgb = humanReadable1QuarterRGBSamples[y % rgbSamplesHeight][x % rgbSamplesWidth];
+        index = (y * testWidth + x) * 3;
+        System.arraycopy(rgb, 0, testSampleBuffer, index, 3);
+      }
+    }
+  }
+
+  protected static final TaskTracker taskTracker = new TaskTracker(ProgressListener.NONE);
+
+  protected Scene createTestScene(
+    int width,
+    int height,
+    int spp,
+    long renderTime
+  ) {
+    Scene scene = new Scene();
+    scene.setCanvasSize(width, height);
+    scene.spp = spp;
+    scene.renderTime = renderTime;
+    return scene;
+  }
+
+  private static final Map<String, String> testDumps = new HashMap<String, String>() {{
```

Comment:
Why not add them as test resource files and load them from there instead of encoding them as strings?

Reply:
This can be done in another PR were JUnit is upgraded and we have a resource folder for tests :)